router/internal/expr/expr.go

```diff
@@ -0,0 +1,169 @@
+package expr
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/expr-lang/expr"
+	"github.com/expr-lang/expr/file"
+	"github.com/expr-lang/expr/vm"
+	"github.com/wundergraph/cosmo/router/pkg/authentication"
+	"net/http"
+	"net/url"
+	"reflect"
+)
+
+// RequestRootContext is the context for expressions parser when evaluating dynamic expressions
+// If possible function calls should be avoided in the expressions as they are much more expensive
+// See https://github.com/expr-lang/expr/issues/734
+type RequestRootContext struct {
```

Comment:
As this is the expression context and we're in the expr package, I'd just call this Context.
As we've discussed in #1481 this context will also contain Response, Field, and other values, so I'd keep it generic.

Reply:
Sounds good to me